File: Backend/Events/Converters/Parties/Digimons/EquipmentsConverter.cs
using Backend.Domain.Models.Parties.Digimons;
using Backend.Events.DTO.Parties.Digimons;

namespace Backend.Events.Converters.Parties.Digimons;

public static class EquipmentsConverter
{
    public static EquipmentsDTO ToDTO(Equipments equipments) => new()
    {
        Head = equipments.Head,
        Body = equipments.Body,
        RightHand = equipments.RightHand,
        LeftHand = equipments.LeftHand,
        Accessory1 = equipments.Accessory1,
        Accessory2 = equipments.Accessory2
    };
}

File: Frontend/src/events/converters/equipments.converter.ts
import type { EquipmentsDTO } from '../events.map';
import type { Equipments } from '../../models';
import { EquipamentConverter } from './equipament.converter';

export class EquipmentsConverter {
    public static convert(equipmentsDto: EquipmentsDTO | null): Equipments {
        return {
            head: EquipamentConverter.convert(equipmentsDto?.head ?? 0),
            body: EquipamentConverter.convert(equipmentsDto?.body ?? 0),
            rightHand: EquipamentConverter.convert(equipmentsDto?.rightHand ?? 0),
            leftHand: EquipamentConverter.convert(equipmentsDto?.leftHand ?? 0),
            accessory1: EquipamentConverter.convert(equipmentsDto?.accessory1 ?? 0),
            accessory2: EquipamentConverter.convert(equipmentsDto?.accessory2 ?? 0)
        };
    }
}
